### PR TITLE
Provide Flink-K8s Runtime Support for StreamX

### PR DESCRIPTION
**Pull request by wolfboys**
Signed-off-by: Al-assad yulin.ying@outlook.com, wolfboys benjobs@qq.com
<br/>
### What problem does this PR solve?

Issue Number: 325

Problem Summary: Provide Flink-K8s Runtime Support for StreamX

### What is changed and how it works?
What's Changed:
* Support for submit/cancel Flink-SQL-Job on Flink-K8s-native session/application mode.
* Support for tracking Flink job status and metrics information from Flink-K8s cluster.
* Support for StreamX instance to manage both Flink-Yarn or Flink-K8s runtime Cluster, and to use either independently.

<br/>

### CODE REVIEW

**Comment by Al-assad** [comment]
LGBT🌝

**Review by Al-assad — CHANGES_REQUESTED** [review]
我们需要要将 pom.xml 中的 streamx 版本更新到 1.2.0

**Review by Al-assad — COMMENTED** [review]
The main code is fine, but we need to update all the streamx-components in pom.xml to version 1.2.0 : )

File: streamx-console/streamx-console-service/src/main/java/com/streamxhub/streamx/console/StreamXConsole.java [hunk 1]
```diff
@@ -28,12 +28,27 @@
...
+ *         \__ \/ __/ ___/ _ \/ __ `/ __ `__ \\ ./
+ *        ___/ / /_/ /  /  __/ /_/ / / / / /  >  <
+ *       /____/\__/_/   \___/\__,_/_/ /_/ /_/_/\ \
+ *                                              \|
+ *
+ *  ┌────────────────────────────────────────────────────┐
+ *  │   WebSite:  http://www.streamxhub.com              │
+ *  │   GitHub :  https://github.com/streamxhub/streamx  │
+ *  │   Gitee  :  https://gitee.com/benjobs/streamx      │
+ *  │   Ver    :  1.1.2                                  │
```
[the remark this answers is not in the document]
Comment (wolfboys) [reply streamx-console/streamx-console-service/src/main/java/com/streamxhub/streamx/console/StreamXConsole.java]: done.

File: streamx-console/streamx-console-service/src/main/resources/banner.txt [hunk 2]
```diff
@@ -1,44 +1,18 @@
...
+             \__ \/ __/ ___/ _ \/ __ `/ __ `__ \\ ./
+            ___/ / /_/ /  /  __/ /_/ / / / / /  >  <
+           /____/\__/_/   \___/\__,_/_/ /_/ /_/_/\ \
+                                                  \|
 
+      ┌────────────────────────────────────────────────────┐
+      │   WebSite:  http://www.streamxhub.com              │
+      │   GitHub :  https://github.com/streamxhub/streamx  │
+      │   Gitee  :  https://gitee.com/benjobs/streamx      │
+      │   Ver    :  1.1.2                                  │
```
[the remark this answers is not in the document]
Comment (wolfboys) [reply streamx-console/streamx-console-service/src/main/resources/banner.txt]: done

File: streamx-flink/streamx-flink-shims/streamx-flink-shims-base/src/main/scala/com/streamxhub/streamx/flink/core/SqlCommandParser.scala [hunk 3]
```diff
@@ -70,35 +67,7 @@ object SqlCommandParser extends Logger {
...
-                }
-                buffer.append(newLine).append("\n")
-              } else {
-                buffer.append(line).append("\n")
-              }
-            }
-            segment.replace(withSegment, buffer.toString.trim)
-          }
-        }
+        groups(i) = matcher.group(i + 1)
```
[the remark this answers is not in the document]
Comment (wolfboys) [reply streamx-flink/streamx-flink-shims/streamx-flink-shims-base/src/main/scala/com/streamxhub/streamx/flink/core/SqlCommandParser.scala]: 👌🏻放到下个版本,重新设计下sql部分.

**Review by Al-assad — APPROVED** [review]
LGTM

**Comment by wolfboys** [comment]
LGTM

**Pull request merged**